src/rocdecode/vaapi/vaapi_videodecoder.cpp

```diff
@@ -592,52 +583,97 @@ void VaapiVideoDecoder::GetDrmNodeOffset(std::string device_name, uint8_t device
     if (!current_compute_partitions.empty()) {
         switch (current_compute_partitions[0]) {
             case kSpx:
-                if (device_id < visible_devices.size()) {
-                    offset = visible_devices[device_id] * 7;
-                } else {
-                    offset = device_id * 7;
-                }
+                offset = 0;
                 break;
             case kDpx:
                 if (device_id < visible_devices.size()) {
-                    offset = (visible_devices[device_id] / 2) * 6;
+                    offset = (visible_devices[device_id] % 2);
```

Comment:
why this is changed

Reply:
This is necessary to accurately calculate the offset for each partition mode on MI300 using the new method. The previous method involved calculating an offset to add to a fixed 128 node.